### ISSUE TITLE
Parameters that are part of the path do not work when specified as character

### ISSUE THREAD

**Issue by ilarischeinin** [trestletech/plumber#273]
There seems to be a minor bug when a parameter that is part of the path is typed as `character`. Here's an example plumber file to demonstrate what I mean:

```r
#* @get /rnorm/<n:int>
#* @param n:int Number of values
function(res, n) {
  rnorm(n)
}

#* @get /broken_bold/<text:character>
#* @param text:character Text to bold
function(res, text) {
  paste0("<b>", text, "</b>")
}

#* @get /working_bold/<text>
#* @param text:character Text to bold
function(res, text) {
  paste0("<b>", text, "</b>")
}
```

All three examples have a parameter that is part of the path. The first endpoint, `/rnorm` has a parameter of type `int` and works as expected.

The second one, `/broken_bold`, has its parameter specified as `character`, but does not work. Swagger identifies the parameter to be part of the query string (instead of path), and the Try it out button results in a request like:
```shell
curl -X GET "http://127.0.0.1:4651/broken_bold/{text}?text=testing" -H "accept: application/json"
```
This results in a 404, as does this:
```shell
curl -X GET "http://127.0.0.1:4651/broken_bold/testing" -H "accept: application/json"
```

But the third endpoint, `/working_bold`, works as expected.

There is thus an easy workaround (just leave out the type from the path), and it is of course redundant to specify the type both in the request path and also in the `@param` specification. Still, it is unexpected that `character` causes an error, whereas integers, numerics, and logicals all work.

**Comment by layik** [comment]
The code [here](https://github.com/trestletech/plumber/blob/f98b41bd639a0dd62906fa30c7799953da2a73e7/R/swagger.R#L10) shows specifying your param type as `character` should be fine.

But I think your second endpoint suffers from this issue here #267. I have a feeling if you remove the type from the @param value it `could` work.

Hope it helps as I did not run your code.

**Comment by ilarischeinin** [comment]
Thanks for the quick reply. Do you mean like this?

```r
#* @get /unsure_bold/<text:character>
#* @param text Text to bold
function(res, text) {
  paste0("<b>", text, "</b>")
}
```

Unfortunately that does not work. Same issue as with the first endpoint.

If I remove the type from both, then I get the error in the issue you linked
```
Error in if (type == "bool" || type == "logical") { : 
  missing value where TRUE/FALSE needed
```

So, with type `character`:

1. query typed, param typed -> doesn't detect that parameter should be part of query
1. query typed, param not typed -> doesn't detect that parameter should be part of query
3. query not typed, param not typed -> errors on `missing value where TRUE/FALSE needed`
4. query not typed, param typed -> works

With type `int`, number 3 above errors out the same way, but all the others work.

**Comment by layik** [comment]
Two things @ilarischeinin 

1. That True/False error, which is the topic in #267 and be aware of it which is as stated in that issue ticket to do with some weird "similar name" issue in the endpoint definition as I understand it, I could be wrong.

2. The endpoint trailing slash issue again discussed here #100 and I am suffering from it which affects all dynamic routes I believe, again I could be wrong.

So this is what I tested for you
```r
#* @get /unsure_bold/<text>
#* @param text Text to bold
function(res, text) {
  paste0("<b>", text, "</b>")
}
```

I removed `character` because it would be converted to `string` anyway, that gives us this

```bash
➜  curl localhost:8000/unsure_bold/helloworld
["<b>helloworld<\/b>"]%                                                               
➜  curl localhost:8000/unsure_bold/    
{"error":["404 - Resource Not Found"]}%                                         
➜  curl localhost:8000/unsure_bold     
{"error":["404 - Resource Not Found"]}%  
```
Now, my understanding and current code works by providing a `default` @param when I make my API calls, I believe this is to do with #100.

I am on plumber `0.4.6`
Hope that helps with moving on.

**Comment by jlfsjunior** [comment]
Any activities on this issue? I am having the same problem here.

**Comment by schloerke** [comment]
Closing issue. Fixed in #351

Not on CRAN as of today.
